ui/src/app/components/take-demographic-survey-v2.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+
+import {
+  DEMOGRAPHIC_SURVEY_V2_NOTIFICATION_END_DATE,
+  DEMOGRAPHIC_SURVEY_V2_PATH,
+} from 'app/utils/constants';
+import { profileStore, serverConfigStore, useStore } from 'app/utils/stores';
+
+import { NotificationBanner } from './notification-banner';
+
+export const TakeDemographicSurveyV2BannerMaybe = () => {
+  const { profile } = useStore(profileStore);
+  const [showBanner, setShowBanner] = useState(true);
+
+  const calculateTimeLeft = () => {
+    const difference =
+      +new Date(DEMOGRAPHIC_SURVEY_V2_NOTIFICATION_END_DATE) - +new Date();
+    let timeLeft = {};
+
+    if (showBanner && difference <= 0) {
+      setShowBanner(false);
```

Comment:
This function has a side-effect (updating state) which can make reasoning about it tricky.  Is there a way to decompose the logic from the state setting?

Reply:
So i might have a way, in the next story i am thinking of keeping the banner as is just change the text and see if that helps.

But as of now i was not able to think of any other way